### PR TITLE
Minimize use of transactions to advance rounds in #360

### PR DESCRIPTION
**Pull request by michaeldiamant**
Extends #360 to propose minimizing use of transactions to advance rounds.

### CODE REVIEW

File: tests/steps/account_v2_steps.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from algosdk import account, constants, encoding, logic
 from algosdk.future import transaction
 from behave import given, then, when
-from tests.steps.other_v2_steps import dev_mode_wait_for_confirmation
+import tests.steps.other_v2_steps  # Imports MaybeString
```
Comment (algochoi) [inline tests/steps/account_v2_steps.py:6]: Good catch

File: tests/steps/account_v2_steps.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def fund_account_address(
     )
     signed_payment = context.wallet.sign_transaction(payment)
     context.app_acl.send_transaction(signed_payment)
-    dev_mode_wait_for_confirmation(context, payment.get_txid(), 10)
+    transaction.wait_for_confirmation(context.app_acl, payment.get_txid(), 10)
```
Comment (algochoi) [inline tests/steps/account_v2_steps.py:21]: I think waiting one round would be sufficient for dev mode:

```suggestion
    transaction.wait_for_confirmation(context.app_acl, payment.get_txid(), 1)
```

Comment (michaeldiamant) [reply tests/steps/account_v2_steps.py]: @algochoi I agree with the suggestion though I think it ought to be applied across-the-board (in other SDKs too).  https://github.com/algorand/py-algorand-sdk/pull/365/commits/a37f2407999dcc04cf9de83a51c377a9c2e1094c brings in the feedback.

Can you resolve if it looks good?

**Review by algochoi — APPROVED** [review]
Thanks for the changes! LGTM

File: tests/steps/application_v2_steps.py [hunk 3]
```diff
@@ -404,21 +403,26 @@ def remember_app_id(context):
     context.app_ids.append(app_id)
 
 
+def wait_for_algod_transaction_processing_to_complete():
+    """
+    wait_wait_for_algod_transaction_processing_to_complete is a Dev mode helper method that's a rough analog to `context.app_acl.status_after_block(last_round + 2)`.
```
Comment (algochoi) [inline tests/steps/application_v2_steps.py:408]: I can change this after merge so we don't run the CI again:
```suggestion
    wait_for_algod_transaction_processing_to_complete is a Dev mode helper method that's a rough analog to `context.app_acl.status_after_block(last_round + 2)`.
```

**Pull request merged**